src/services/siwe/index.ts

```diff
@@ -38,4 +40,28 @@ Issued At: ${message.issuedAt.toISOString()}`
   return verifyAuth({ message: signableMessage, signature })
 }
 
-export default signInWithEthereum
+const createUserAccount = async (address: string) => {
+  return await fetch(`${GATEWAY_URL_STAGING}/v1/accounts`, {
+    method: 'POST',
+    headers: {
+      accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+    credentials: 'include',
+    body: JSON.stringify({ address }),
+  }).then((response) => response.json())
+}
+
+export const getUserAccount = async (address: string) => {
+  if (!address) return
+
+  return fetch(`${GATEWAY_URL_STAGING}/v1/accounts/${address}`, { credentials: 'include' }).then((res) => {
+    if (res.ok && res.status === 200) {
+      return res.json()
+    } else if (res.status === 404) {
+      return createUserAccount(address)
+    } else {
+      throw new Error('Error fetching User Account')
+    }
+  })
+}
```

Comment:
We can use the latest version 3.22.1 of the typescript-gateway-sdk which contains these functions.

Reply:
oh nice, thanks 👍